WanakuServiceCatalog reconciler cannot authenticate with OIDC-protected router

## Description

The `WanakuServiceCatalogReconciler` fails to deploy catalogs to an OIDC-protected router. It receives HTTP 302 (redirect to OIDC login) instead of authenticating.

## Steps to Reproduce

1. Deploy a WanakuRouter with Keycloak auth enabled
2. Create a WanakuServiceCatalog CR referencing a ConfigMap with catalog data
3. Observe the reconciler failing with HTTP 302

## Expected Behavior

The operator should authenticate with the router using OIDC credentials before deploying service catalogs.

## Actual Behavior

```
WanakuException: Failed to deploy service catalog 'hello-system': HTTP 302 -
Caused by: ClientWebApplicationException: Received: 'HTTP 302 Found'
  when invoking REST Client method: 'ai.wanaku.core.services.api.ServiceCatalogService#deploy'
```

## Additional Context

The `WanakuCamelRouteReconciler` (in source) appears to handle OIDC auth, but the `WanakuServiceCatalogReconciler` does not.

--- a/apps/wanaku-operator/deploy/helm/wanaku-operator/templates/deployment.yaml
+++ b/apps/wanaku-operator/deploy/helm/wanaku-operator/templates/deployment.yaml
@@ -32,6 +32,12 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.namespace
+            - name: WANAKU_OIDC_CLIENT_SECRET
+              valueFrom:
+                secretKeyRef:
+                  key: client-secret
+                  name: {{ .Values.app.oidc.secretName }}
+                  optional: true
             - name: QUARKUS_OPERATOR_SDK_CONTROLLERS_WANAKU_CAPABILITY_NAMESPACES
               value: {{ .Values.app.envs.QUARKUS_OPERATOR_SDK_CONTROLLERS_WANAKU_CAPABILITY_NAMESPACES | quote }}
             - name: QUARKUS_OPERATOR_SDK_CONTROLLERS_WANAKU_CAMEL_ROUTE_NAMESPACES

--- a/apps/wanaku-operator/deploy/helm/wanaku-operator/values.yaml
+++ b/apps/wanaku-operator/deploy/helm/wanaku-operator/values.yaml
@@ -11,6 +11,8 @@ app:
     QUARKUS_OPERATOR_SDK_CONTROLLERS_WANAKU_ROUTER_NAMESPACES: JOSDK_WATCH_CURRENT
     QUARKUS_OPERATOR_SDK_CONTROLLERS_WANAKU_SERVICE_CATALOG_NAMESPACES: JOSDK_WATCH_CURRENT
   image: quay.io/wanaku/wanaku-operator:latest
+  oidc:
+    secretName: wanaku-oidc
   imagePullPolicy: IfNotPresent
   livenessProbe:
     failureThreshold: 3

--- a/apps/wanaku-operator/src/main/helm/values.yaml
+++ b/apps/wanaku-operator/src/main/helm/values.yaml
@@ -8,3 +8,5 @@ app:
     capabilities:
       drop:
         - ALL
+  oidc:
+    secretName: wanaku-oidc

--- a/apps/wanaku-operator/src/main/resources/application.properties
+++ b/apps/wanaku-operator/src/main/resources/application.properties
@@ -64,3 +64,11 @@ quarkus.helm.values."app.imagePullPolicy".paths=(kind == Deployment).spec.templa
 quarkus.kubernetes.security-context.run-as-non-root=true
 quarkus.helm.values."podSecurityContext.runAsNonRoot".paths=(kind == Deployment).spec.template.spec.securityContext.runAsNonRoot
 quarkus.helm.values."podSecurityContext.runAsNonRoot".description=Require the pod to run as a non-root user.
+
+# OIDC client secret sourced from a Kubernetes Secret
+quarkus.kubernetes.env.mapping.wanaku-oidc-client-secret.from-secret=wanaku-oidc
+quarkus.kubernetes.env.mapping.wanaku-oidc-client-secret.with-key=client-secret
+
+# Make the OIDC Secret name configurable via Helm values
+quarkus.helm.values."app.oidc.secretName".paths=(kind == Deployment).spec.template.spec.containers.env.(name == WANAKU_OIDC_CLIENT_SECRET).valueFrom.secretKeyRef.name
+quarkus.helm.values."app.oidc.secretName".description=Name of the Kubernetes Secret containing the OIDC client secret.

